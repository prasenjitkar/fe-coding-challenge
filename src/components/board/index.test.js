import React from 'react';
import { render, screen } from '@testing-library/react';
import { Board } from '.';
import configureStore from '../../store';
import { Provider } from 'react-redux';

const mockData = { board: [["X", "X", "X"],[null, null, null],[null, null, null]], game: {currentPlayer: "X", winner: "X"}};

describe('game', () => {
test('renders Board without mock data', () => {
  const  { asFragment } = render(<Provider store={configureStore()}><Board /></Provider>);
  const boardText = screen.getByText(/Player/i);
  expect(boardText).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('renders Board with winner state', () => {
  const  { asFragment } = render(<Provider store={configureStore(mockData)}><Board /></Provider>);
  const boardText = screen.getByText(/Winner/i);
  expect(boardText).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
});
