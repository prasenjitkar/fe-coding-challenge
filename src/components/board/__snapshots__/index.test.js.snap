// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game renders Board with winner state 1`] = `
<DocumentFragment>
  <div
    class="game"
  >
    <div
      class="tic-tac-toe-board"
    >
      <div
        class="row"
      >
        <div
          class="cell"
        >
          X
        </div>
        <div
          class="cell"
        >
          X
        </div>
        <div
          class="cell"
        >
          X
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="cell"
        />
        <div
          class="cell"
        />
        <div
          class="cell"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="cell"
        />
        <div
          class="cell"
        />
        <div
          class="cell"
        />
      </div>
    </div>
    <div
      class="w3-container w3-center w3-animate-top"
    >
      <span
        class="winner"
      >
        Winner: X 
      </span>
    </div>
    <div
      class="restart-button-section"
    >
      <button
        class="restart-button"
      >
        Restart
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`game renders Board without mock data 1`] = `
<DocumentFragment>
  <div
    class="game"
  >
    <div
      class="tic-tac-toe-board"
    >
      <div
        class="row"
      >
        <div
          class="cell"
        />
        <div
          class="cell"
        />
        <div
          class="cell"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="cell"
        />
        <div
          class="cell"
        />
        <div
          class="cell"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="cell"
        />
        <div
          class="cell"
        />
        <div
          class="cell"
        />
      </div>
    </div>
    <div
      class="player"
    >
      Player X 
    </div>
    <div
      class="restart-button-section"
    >
      <button
        class="restart-button"
        disabled=""
      >
        Restart
      </button>
    </div>
  </div>
</DocumentFragment>
`;
